repo_policy_compliance/github_client.py

```diff
@@ -81,7 +82,11 @@ def wrapper(*args: P.args, **kwargs: P.kwargs) -> R:
                 "please wait before retrying."
             ) from exc
         except GithubException as exc:
-            raise GithubClientError("The github client encountered an error.") from exc
+            logging.error("Github client error: %s", exc, exc_info=exc)
+            raise GithubClientError(
+                "Something went wrong checking Github repository policy compliance. Please contact "
+                "the operator."
+            ) from exc
```

Comment:
Mhh, having the "end user" message here does not seem right, as it is not the concern of the `github_client` module. Personally, I prefer the way you implemented it before (explicitly setting the end-user error message as the reason in `check.py`).

Reply:
The problem with that is that this module abstracts away all github errors into a single github client error. Perhaps it would be better if this module didnt catch the errors and the errors would be handled by the wrapper of the outer module. What do you think?

Reply:
The reason i prefer to not abstract the errors away is to save the RateLimitExceeded error, which is retryable on the user end without the operator involvement.
The best I can think of right now is to further introduce a new error inheriting the GithubClientError, perhaps a RetryableGithubError to indicate an action that can be taken from the user end as well.

Reply:
The point of the github module is to abstract the details of handling GitHub so it should be remapping exceptions. I don't think this module should be aware it is running in the GitHub runner context, that should be reserved for the caller to interpret the exception. You could introduce new exceptions to distinguish between non-retryable and retryable errors which can then be used by the caller to decide what to do

Reply:
Introduced a new `RetryableGithubClientError` which can pass on the message to the user.